controllers/packetcluster_controller.go

```diff
@@ -151,17 +153,17 @@ func (r *PacketClusterReconciler) reconcileNormal(ctx context.Context, clusterSc
 	if clusterScope.PacketCluster.Spec.VIPManager == "KUBE_VIP" {
 		if err := r.PacketClient.EnableProjectBGP(ctx, packetCluster.Spec.ProjectID); err != nil {
 			log.Error(err, "error enabling bgp for project")
-			return ctrl.Result{}, err
+			return err
 		}
 	}
 
 	clusterScope.PacketCluster.Status.Ready = true
 	conditions.MarkTrue(packetCluster, infrav1.NetworkInfrastructureReadyCondition)
 
-	return ctrl.Result{}, nil
+	return nil
 }
 
-func (r *PacketClusterReconciler) reconcileDelete(ctx context.Context, clusterScope *scope.ClusterScope) (ctrl.Result, error) {
+func (r *PacketClusterReconciler) reconcileDelete(_ context.Context, _ *scope.ClusterScope) (ctrl.Result, error) {
 	// Initially I created this handler to remove an elastic IP when a cluster
```

Comment:
rereading this for the first time in a while.. We probably should clean up after ourselves for any resources we dynamically provisioned. We could offer preservation behaviors in a few different ways (possibly all of them):
- IP reservation tags `locked` (? this wouldn't be conventional for IP reservations. Unlike devices there is no "locked" field on the EM API resource. most likely this would look like other CAPP EM tags)
- CAPP configuration parameters `--no-ip-reservation-cleanup`
- PacketCluster configuration parameters

#617 
#226